include/cuda/std/detail/libcxx/include/__type_traits/is_pointer.h

```diff
@@ -25,7 +25,7 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-#if defined(_LIBCUDACXX_IS_POINTER) && !defined(_LIBCUDACXX_USE_IS_POINTER_FALLBACK)
+#if 0 // defined(_LIBCUDACXX_IS_POINTER) && !defined(_LIBCUDACXX_USE_IS_POINTER_FALLBACK)
```

Comment:
Uhh, so what's up with the `#if 0`s here and in `is_scalar`, `is_signed`, and `is_void`? The commit for them says "fixes for Clang", does it mean that the final version of this will instead correctly set the fallback macro for the problematic versions of Clang, or is this just a dirty patch?

Reply:
So this is something really painful here. The code is factually correct. We check for the existence of a given builtin and call it. https://godbolt.org/z/hTh7aeEzs

However, if I actually try to build that code with nvcc and clang (and only clang), then I am getting:
```
/home/miscco/libcudacxx/include/cuda/std/detail/libcxx/include/__functional/../__utility/../__type_traits/../__type_traits/is_pointer.h:31:113: error: '_Tp' does not refer to a value
struct __attribute((__type_visibility__("default" ))) is_pointer : public integral_constant< bool, __is_pointer(_Tp)>  { 
                                                                                                                ^
/home/miscco/libcudacxx/include/cuda/std/detail/libcxx/include/__functional/../__utility/../__type_traits/../__type_traits/is_pointer.h:30:17: note: declared here
template< class _Tp> 
```

I should add a todo to file a compiler bug

Reply:
We should do what I suggested then - in `__config`, specifically for the combination of nvcc and clang, define `_LIBCUDACXX_USE_..._FALLBACK` for the four broken ones, and in the future when it's fixed we will guard defining that macro with a version check against (presumably) nvcc.